ios/Components/InstantComponent.swift

```diff
@@ -0,0 +1,155 @@
+//
+// Copyright (c) 2022 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+
+import Adyen
+import Foundation
+import PassKit
+import React
+
+@objc(AdyenInstant)
+internal class InstantComponent: BaseModule {
+    
+    private var currentComponent: PresentableComponent?
+    private var currentPaymentComponent: PaymentComponent?
+    private var actionHandler: AdyenActionComponent?
+    
+    @objc
+    override static func requiresMainQueueSetup() -> Bool { true }
+    override func stopObserving() {}
+    override func startObserving() {}
+    override func supportedEvents() -> [String]! { super.supportedEvents() }
+
+    @objc
+    func hide(_ success: NSNumber, event: NSDictionary) {
+        DispatchQueue.main.async {
+            self.currentComponent?.finalizeIfNeeded(with: success.boolValue)
+            UIApplication.shared.keyWindow?.rootViewController?.dismiss(animated: true)
+
+            self.actionHandler = nil
+            self.currentComponent = nil
+            self.currentPaymentComponent = nil
+        }
+    }
+
+    @objc
+    func open(_ paymentMethods: NSDictionary, configuration: NSDictionary) {
+        let paymentMethod: PaymentMethod
+        guard let data = try? JSONSerialization.data(withJSONObject: paymentMethods, options: []) else {
```

Comment:
why `JSONSerialization `?

Reply:
Because we receive `NSDictionary` from RN but I prefere to parse it as `Decodable` - don't want to create new decoder for each class